plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergSystemTables.java

```diff
@@ -364,6 +382,121 @@ public void testFilesTable()
+                "SELECT (CAST(ROW(123) AS ROW(y int)))",
+                "{\"x.y\":{\"column_size\":" + columnSize(37) + ",\"value_count\":1,\"null_value_count\":0,\"nan_value_count\":null,\"lower_bound\":123,\"upper_bound\":123}}");
+        testFilesTableReadableMetrics(
+                "array(int)",
+                "VALUES ARRAY[123]",
+                "{\"x.element\":{\"column_size\":" + columnSize(43) + ",\"value_count\":" + value(1, null) + ",\"null_value_count\":" + value(0, null) + ",\"nan_value_count\":null,\"lower_bound\":null,\"upper_bound\":null}}");
+        testFilesTableReadableMetrics(
+                "map(int, int)",
+                "VALUES map(ARRAY[1,3], ARRAY[2,4])",
+                "{" +
+                        "\"x.key\":{\"column_size\":" + columnSize(47) + ",\"value_count\":" + value(2, null) + ",\"null_value_count\":" + value(0, null) + ",\"nan_value_count\":null,\"lower_bound\":null,\"upper_bound\":null}," +
+                        "\"x.value\":{\"column_size\":" + columnSize(47) + ",\"value_count\":" + value(2, null) + ",\"null_value_count\":" + value(0, null) + ",\"nan_value_count\":null,\"lower_bound\":null,\"upper_bound\":null}" +
+                        "}");
+    }
+
+    private void testFilesTableReadableMetrics(@Language("SQL") String type, @Language("SQL") String values, @Language("JSON") String... readableMetrics)
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_files_table", "(x " + type + ")")) {
+            getQueryRunner().execute("INSERT INTO " + table.getName() + " " + values);
+            assertThat(computeActual("SELECT readable_metrics FROM \"" + table.getName() + "$files\"").getOnlyColumnAsSet())
+                    .containsExactlyInAnyOrder(readableMetrics);
+        }
+    }
+
+    @Test
+    public void testFilesSchemaEvolution()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_files_table", "WITH (partitioning = ARRAY['part']) AS SELECT 1 x, 2 part")) {
+            assertThat(query("SELECT partition FROM \"" + table.getName() + "$files\""))
+                    .matches("SELECT CAST(ROW(2) AS ROW(part int))");
+
+            assertUpdate("ALTER TABLE " + table.getName() + " ADD COLUMN another_part int");
+            assertUpdate("ALTER TABLE " + table.getName() + " SET PROPERTIES partitioning = ARRAY['part', 'another_part']");
+            assertThat(query("SELECT partition FROM \"" + table.getName() + "$files\""))
+                    .matches("SELECT CAST(ROW(2, NULL) AS ROW(part int, another_part int))");
+
+            assertUpdate("ALTER TABLE " + table.getName() + " RENAME COLUMN x TO part_renamed");
+            assertThat(query("SELECT partition FROM \"" + table.getName() + "$files\""))
+                    .matches("SELECT CAST(ROW(2, NULL) AS ROW(part int, another_part int))");
+
+            assertUpdate("ALTER TABLE " + table.getName() + " DROP COLUMN part_renamed");
```

Comment:
what happens when you rename part column (for new and old file)?

Reply:
It shows the original column names. That is the Spark behavior:
```sql
spark-sql (default)> CREATE TABLE test(x int, part int) USING iceberg PARTITIONED BY (part);
spark-sql (default)> INSERT INTO test VALUES (1, 10);
spark-sql (default)> ALTER TABLE test RENAME COLUMN part to renamed;
spark-sql (default)> SELECT partition FROM iceberg_test.default.test.files;
{"part":10}
spark-sql (default)> INSERT INTO test VALUES (2, 20);
spark-sql (default)> SELECT partition FROM iceberg_test.default.test.files;
{"part":20}
{"part":10}
```